assets/js/index.js

```diff
@@ -147,14 +163,46 @@ function bindDeleteEvent() {
 }
 
 function bindUpdateEvent() {
-  const checkboxs = Array.from(document.getElementsByClassName("checkbox"));
-  checkboxs.forEach(checkbox => {
+  const checkboxes = Array.from(document.getElementsByClassName("checkbox"));
+  checkboxes.forEach(checkbox => {
     checkbox.addEventListener("change", function (event) {
       updateStatus(event.target.parentNode.id, this.checked);
     })
   });
 }
 
+function bindEditEvent() {
+  const editIcons = Array.from(document.getElementsByClassName("edit-icon"));
+  editIcons.forEach(editIcon => {
+    editIcon.addEventListener("click", function (event) {
+      const taskId = event.target.parentNode.id;
+      const todoContainer = document.getElementById(`todo-container-${taskId}`);
+      const todoText = todoContainer.innerText;
+      todoContainer.innerHTML = `<input id="input" class="edited-todo" type="text" value="${todoText}"/>
+      <i id="check-icon-${taskId}" class="fa-solid fa-check check-icon"></i>`
+      const checkIcon = document.getElementById(`check-icon-${taskId}`);
+      const input = document.getElementById("input");
+      checkIcon.addEventListener("click", function () {
+        if (input.value === "") {
+          return todoContainer.innerText = todoText;
+        }
+        if (input.value !== todoText) {
+          editTodo(taskId, input.value);
+        } else {
+          renderTasks();
+        }
+      })
+      handleIcons(taskId);
```

Comment:
I'm not sure about how this function is used. It does not hide anything 🤔

Reply:
I sent you a voice message.